packages/utilities/src/hooks/useViewLoading.ts

```diff
@@ -1,16 +1,34 @@
 import { useEffect, useState } from 'react';
 
-export default function useViewLoading(view: __esri.MapView | null) {
+let timeoutId: NodeJS.Timeout | null = null;
+
+export default function useViewLoading(
+  view: __esri.MapView | null,
+  debounceDuration = 500,
+) {
   const [isLoading, setIsLoading] = useState(false);
 
   useEffect(() => {
-    const init = () => {
-      view?.watch('updating', (updating: boolean) => setIsLoading(updating));
-    };
-
-    if (view) {
-      init();
+    if (!view) {
+      return;
     }
+
+    view.when(() => {
+      view.watch('updating', (updating: boolean) => {
+        if (timeoutId) {
+          return;
+        }
+
+        if (updating) {
+          setIsLoading(true);
+        } else {
+          timeoutId = setTimeout(() => {
+            setIsLoading(false);
+            timeoutId = null;
+          }, debounceDuration);
```

Comment:
```suggestion
        if (updating && timeoutId) {
          return;
        }

        if (!updating) {
          if (timeoutId) {
            clearTimeout(timeoutId);
            timeoutId = null;
          }
          setIsLoading(false);
        } else {
          timeoutId = setTimeout(() => {
            setIsLoading(true);
            timeoutId = null;
          }, debounceDuration);
```

I think that I would flip this from quick to show and slow to hide to slow to show and quick to hide so that the loader only shows up if the map is updating for longer than the debounce time. This resulted in the loader not being show as much during times like zooming and panning the map in my testing.